torchtitan/distributed/utils.py

```diff
@@ -202,6 +204,30 @@ def context(cp_context: Generator[None, None, None] | None = None):
     return context
 
 
+def maybe_enable_amp(
+    parallel_dims: ParallelDims, mixed_precision_param: str, device_type: torch.device
+) -> Generator[None, None, None]:
+    fsdp_enabled = parallel_dims.dp_shard_enabled or parallel_dims.cp_enabled
```

Comment:
Can we make this a property of `ParallelDims`? It seems that this is pretty commonly used.